pkg/ccl/changefeedccl/helpers_test.go

```diff
@@ -396,6 +382,186 @@ func cloudStorageTest(
 	}
 }
 
+type fakeKafkaClient struct{}
+
+func (c *fakeKafkaClient) Partitions(topic string) ([]int32, error) {
+	return []int32{0}, nil
+}
+
+func (c *fakeKafkaClient) RefreshMetadata(topics ...string) error {
+	return nil
+}
+
+func (c *fakeKafkaClient) Close() error {
+	return nil
+}
+
+var _ kafkaClient = (*fakeKafkaClient)(nil)
+
+type ignoreCloseProducer struct {
+	*asyncProducerMock
+}
+
+func (p *ignoreCloseProducer) Close() error {
+	return nil
+}
+
+// teeGroup facilitates reading messages from input channel
+// and sending them to one or more output channels.
+type teeGroup struct {
+	g    ctxgroup.Group
+	done chan struct{}
+}
+
+func newTeeGroup() *teeGroup {
+	return &teeGroup{
+		g:    ctxgroup.WithContext(context.Background()),
+		done: make(chan struct{}),
+	}
+}
+
+// tee reads incoming messages from input channel and sends them out to one or more output channels.
+func (tg *teeGroup) tee(in <-chan *sarama.ProducerMessage, out ...chan<- *sarama.ProducerMessage) {
+	tg.g.Go(func() error {
+		for {
+			select {
+			case <-tg.done:
+				return nil
+			case m := <-in:
+				for i := range out {
+					select {
+					case <-tg.done:
+						return nil
+					case out[i] <- m:
+					}
+				}
+			}
+		}
+	})
+}
+
+// wait shuts down tee group.
+func (tg *teeGroup) wait() error {
+	close(tg.done)
+	return tg.g.Wait()
+}
+
+func watchdog(timeout time.Duration) (cancel func(), wait func()) {
+	var wg sync.WaitGroup
+	wg.Add(1)
+	cancelCh := make(chan struct{})
+	go func() {
+		defer wg.Done()
+		select {
+		case <-time.After(timeout):
+			panic(fmt.Sprintf("test timeout\n%s", debug.Stack()))
```

Comment:
I'm a little worried about this because I know we had a recent issue with a panic in a goroutine stopping the entire test runner meaning that the nightlies had a lot of tests that weren't actually being run.